packages/password-input/src/PasswordInput/PasswordInput.spec.tsx

```diff
@@ -0,0 +1,230 @@
+      const { labelElement } = renderPasswordInput({
+        label: defaultProps.label,
+      });
+      expect(labelElement?.innerHTML).toContain(defaultProps.label);
+    });
+
+    test('renders no label tag when label is not supplied', () => {
+      renderPasswordInput();
+      expect(screen.queryByRole('label')).not.toBeInTheDocument();
+    });
+
+    test(`renders ${defaultProps.placeholder} as placeholder text`, () => {
+      const { getByPlaceholderText } = renderPasswordInput({
+        placeholder: defaultProps.placeholder,
+      });
+      expect(getByPlaceholderText(defaultProps.placeholder)).toBeVisible();
+    });
+
+    test(`renders ${defaultProps.className} in the classList`, () => {
+      const { container } = renderPasswordInput({
+        className: defaultProps.className,
+      });
+      expect(
+        (container?.firstChild as HTMLElement)?.classList.contains(
+          defaultProps.className,
+        ),
+      ).toBe(true);
+    });
+
+    test('input id attribute matches id prop', async () => {
+      const { passwordInput } = renderPasswordInput({
+        id: defaultProps.id,
+      });
+      expect(passwordInput.getAttribute('id')).toBe(defaultProps.id);
+    });
+
+    test('renders ${defaultProps.autoComplete} as autoComplete text', async () => {
+      const { passwordInput } = renderPasswordInput({
+        autoComplete: defaultProps.autoComplete,
+      });
+      expect(passwordInput.getAttribute('autoComplete')).toBe(
+        defaultProps.autoComplete,
+      );
+    });
+
+    test('value change triggers onChange callback', () => {
+      const { passwordInput } = renderPasswordInput({
+        label: defaultProps.label,
+        onChange: defaultProps.onChange,
+      });
+
+      expect((passwordInput as HTMLInputElement).value).toBe('');
+
+      fireEvent.change(passwordInput, {
+        target: { value: 'a' },
+      });
+
+      expect((passwordInput as HTMLInputElement).value).toBe('a');
+      expect(defaultProps.onChange).toHaveBeenCalledTimes(1);
+    });
+
+    test('blur triggers onBlur callback', () => {
+      const { passwordInput } = renderPasswordInput({
+        label: defaultProps.label,
+        onBlur: defaultProps.onBlur,
+      });
+
+      userEvent.tab(); // focus
+      expect(passwordInput).toHaveFocus();
+      userEvent.tab(); // blur
+
+      expect(defaultProps.onBlur).toHaveBeenCalledTimes(1);
+    });
+
+    test('toggles password type on click of toggle icon ', () => {
+      const { passwordInput, toggleButton } = renderPasswordInput();
+
+      expect(passwordInput.getAttribute('type')).toBe('password');
+      fireEvent.click(toggleButton as HTMLButtonElement);
+      expect(passwordInput.getAttribute('type')).toBe('text');
+      fireEvent.click(toggleButton as HTMLButtonElement);
+      expect(passwordInput.getAttribute('type')).toBe('password');
+    });
+  });
+
+  describe('Password input with stateNotifications array', () => {
```

Comment:
can we add test cases to make sure that the stateNotifications are rendered with the proper icon based on state?

Reply:
yes!

Reply:
added to stateNotifications component test